pkg/function/backup_data_using_kopia_server.go

```diff
@@ -0,0 +1,260 @@
+	configFile, logDirectory := kankopia.GetCustomConfigFileAndLogDirectory(hostname)
+
+	cmd := kopiacmd.RepositoryConnectServerCommand(kopiacmd.RepositoryServerCommandArgs{
+		UserPassword:    userPassphrase,
+		ConfigFilePath:  configFile,
+		LogDirectory:    logDirectory,
+		CacheDirectory:  kopiacmd.DefaultCacheDirectory,
+		Hostname:        hostname,
+		ServerURL:       serverAddress,
+		Fingerprint:     fingerprint,
+		Username:        username,
+		ContentCacheMB:  contentCacheMB,
+		MetadataCacheMB: metadataCacheMB,
+	})
+
+	stdout, stderr, err := kube.Exec(cli, namespace, pod, container, cmd, nil)
+	format.Log(pod, container, stdout)
+	format.Log(pod, container, stderr)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to connect to Kopia Repository Server")
+	}
+
+	cmd = kopiacmd.SnapshotCreate(
+		kopiacmd.SnapshotCreateCommandArgs{
+			PathToBackup: includePath,
+			CommandArgs: &kopiacmd.CommandArgs{
+				RepoPassword:   "",
+				ConfigFilePath: configFile,
+				LogDirectory:   logDirectory,
+			},
+			Tags:                   tags,
+			ProgressUpdateInterval: 0,
+		})
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to generate snapshot create command")
+	}
+	stdout, stderr, err = kube.Exec(cli, namespace, pod, container, cmd, nil)
+	format.Log(pod, container, stdout)
+	format.Log(pod, container, stderr)
+
+	message := "Failed to create and upload backup"
+	if err != nil {
+		if strings.Contains(err.Error(), kerrors.ErrCodeOutOfMemoryStr) {
+			message = message + ": " + kerrors.ErrOutOfMemoryStr
+		}
+		return nil, errors.Wrap(err, message)
+	}
+	// Parse logs and return snapshot IDs and stats
+	return kopiacmd.ParseSnapshotCreateOutput(stdout, stderr)
+}
+
+func getHostNameAndUserPassPhraseFromRepoServer(userCreds string) (string, string, error) {
+	var userAccessMap map[string]string
+	if err := json.Unmarshal([]byte(userCreds), &userAccessMap); err != nil {
+		return "", "", errors.Wrap(err, "Failed to unmarshal User Credentials Data")
+	}
+
+	var userPassPhrase string
+	var hostName string
+	for key, val := range userAccessMap {
+		hostName = key
+		userPassPhrase = val
+	}
+
+	decodedUserPassPhrase, err := base64.StdEncoding.DecodeString(userPassPhrase)
+	if err != nil {
+		return "", "", errors.Wrap(err, "Failed to Decode User Passphrase")
+	}
+	return hostName, string(decodedUserPassPhrase), nil
+
+}
+
+func getRepositoryServerAddress(cli kubernetes.Interface, rs param.RepositoryServer) (string, error) {
+	repositoryServerService, err := cli.CoreV1().Services(rs.Namespace).Get(context.Background(), rs.ServerInfo.ServiceName, metav1.GetOptions{})
+	if err != nil {
+		return "", errors.New("Unable to find Service Details for Repository Server")
+	}
+	repositoryServerServicePort := strconv.Itoa(int(repositoryServerService.Spec.Ports[0].Port))
+	serverAddress := fmt.Sprintf("https://%s.%s.svc.cluster.local:%s", rs.ServerInfo.ServiceName, rs.Namespace, repositoryServerServicePort)
```

Comment:
```suggestion

	serverAddress := fmt.Sprintf("https://%s.%s.svc.cluster.local:%d", rs.ServerInfo.ServiceName, rs.Namespace, repositoryServerService.Spec.Ports[0].Port)
```

would this not work?

Reply:
It would work, made the changes !